tools/pipelines-tasks/test/azure-arm-rest-mock.ts

```diff
@@ -0,0 +1,92 @@
+import path = require('path');
+import fs = require('fs');
+import { ClientSecretCredential } from "@azure/identity";
+import { TokenCredential } from "@azure/core-auth";
+import { StorageManagementClient as CustomStorageClient}  from '@azure/arm-storage';
+
+let bearerAccessToken: string | undefined;
+let credential: TokenCredential;
+
+export class AzureRMEndpoint {
+    private _connectedServiceName: any;
+    private endpoint: any;
+    public subscriptionID: string;
+    constructor(connectedServiceName: any) {
+        this._connectedServiceName = connectedServiceName;
+        this.endpoint = new AzureEndpoint();
+        this.subscriptionID = connectedServiceName;
+    }
+
+    getEndpoint() {
+        this.endpoint.applicationTokenCredentials = new ApplicationTokenCredentials();
+        this.endpoint.subscriptionID = this._connectedServiceName;
+        return this.endpoint;
+    }
+}
+
+export class AzureEndpoint {
+    public applicationTokenCredentials: any;
+    public subscriptionID: any;
+}
+
+export class ApplicationTokenCredentials {
+    async getToken() {
+        try {
+            credential = new ClientSecretCredential(tenantId, clientId, clientSecret);
+            const armResourceEndpoint = 'https://management.azure.com/.default';
```

Comment:
So the tests are going to be running against the actual endpoints, right? Do these tests end up creating anything in Azure that needs to be cleaned up?
Do you have any (internal) documentation of what parameters are to be used in the pipelines and how to set them?

Reply:
The clean has to be done manual, this test just publishes the test application to the the provided azure or avd account.